### ISSUE TITLE
Can loss be computed on multiple gpus?

### ISSUE THREAD

**Issue by Mistariano** [keras-team/keras#12280]
Hi guys. I'm using `multi_gpu_model` and I found that all of the losses were computed on the first gpu only.
So It can easily cause an OOM error when I set a larger batch-size, while other gpus didn't work efficiently.

I'm using `multi_gpu_model` likes this:
```
with tf.device('/cpu:0'):
    model = build_model()

parallel_model = multi_gpu_model(model)
parallel_model.compile(my_loss_func, 'adam')

parallel_model.fit(...)
model.save(...)
```

where `my_loss_func` is a very complex loss function using outputs of a pretrained VGG model.

I set a tensorflow session with `log_device_placement=True` and found the problem.

I'd like to separate the loss computing, but I have no idea. I think it may be a new feature of `multi_gpu_model` to support computing loss on multiple gpus, but I'm not sure whether I made any mistake.

Looking forward to your advice and opinion. Thanks. :)
